phoenix-core/src/it/java/org/apache/phoenix/end2end/index/ShortViewIndexIdIT.java

```diff
@@ -0,0 +1,104 @@
+        Map<String, String> clientProps = Maps.newHashMapWithExpectedSize(1);
+        clientProps.put(QueryServices.LONG_VIEW_INDEX_ENABLED_ATTRIB, "false");
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()), new ReadOnlyProps(clientProps.entrySet().iterator()));
+    }
+
+    @Test
+    public void testCreateLocalIndexWithData() throws Exception {
+        String tableName = generateUniqueName();
+        String indexName = "IDX_" + generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement()
+                    .execute("CREATE TABLE " + tableName + " (pk INTEGER PRIMARY KEY, v INTEGER)");
+            conn.commit();
+            conn.createStatement().execute("UPSERT INTO " + tableName + " VALUES(1, 1)");
+            conn.commit();
+            conn.createStatement()
+                    .execute("CREATE LOCAL INDEX " + indexName + " ON " + tableName + "(v)");
+            conn.commit();
+            // this should return the expected data if the index was written correctly
+            ResultSet rs = conn.createStatement()
+                    .executeQuery("SELECT v FROM " + tableName + " WHERE v < 2");
+            assertTrue(rs.next());
+            assertEquals(1, rs.getInt(1));
+            rs.close();
+        }
+    }
+
+    @Test
+    public void testUpsertIntoLocalIndex() throws Exception {
+        String tableName = generateUniqueName();
+        String index1 = "IDX_" + generateUniqueName();
+        String index2 = "IDX_" + generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement()
+                    .execute("CREATE TABLE " + tableName + " (pk INTEGER PRIMARY KEY, v1 INTEGER, v2 INTEGER)");
+            conn.createStatement()
+            .execute("CREATE LOCAL INDEX " + index1 + " ON " + tableName + "(v1)");
+            conn.createStatement()
+            .execute("CREATE LOCAL INDEX " + index2 + " ON " + tableName + "(v2)");
+            conn.commit();
+            conn.createStatement().execute("UPSERT INTO " + tableName + " VALUES(1, 2, 3)");
+            conn.commit();
+            ResultSet rs = conn.createStatement()
+                    .executeQuery("SELECT * FROM " + index1);
+            // we're expecting exactly one row mapping the column value to the key
+            assertTrue(rs.next());
+            assertEquals(2, rs.getInt(1));
+            assertEquals(1, rs.getInt(2));
+            assertFalse(rs.next());
+            rs.close();
+
+            rs = conn.createStatement()
+                    .executeQuery("SELECT * FROM " + index2);
+            // we're expecting exactly one row mapping the column value to the key
+            assertTrue(rs.next());
+            assertEquals(3, rs.getInt(1));
+            assertEquals(1, rs.getInt(2));
+            assertFalse(rs.next());
+            rs.close();
+        }
+    }
+
+}
```

Comment:
Any way we can add a test for the stale region boundaries change as well?

Reply:
That's tested in MutableIndexSplitForwardScanIT and MutableIndexSplitReverseScanIT. Those tests fails with the first version of the change.